api/v1/controllers/subjects.js

```diff
@@ -326,8 +329,10 @@ module.exports = {
     } else {
       doGetHierarchy(resultObj)
       .then((resultObj) => {
-        u.logAPI(req, resultObj, resultObj.retval);
-        res.status(httpStatus.OK).json(resultObj.retval);
+        const subject = resultObj.retval;
+        u.removeFieldsFromResponse(opts.fieldsToExclude, subject);
```

Comment:
This is only going to remove the fields from the top-level subject in the hierarchy, right? What about n-depth children?

Reply:
It's just removing from the top-level. Seems that this methods is always removing the root object. Even when there is a list of objects, it iterates and removes attributes from the top-level.

DoFind.js, method doFindAndCountAll:
```
for (let i = retval.length - 1; i >= 0; i--) {
 u.removeFieldsFromResponse(fieldsToExclude, retval[i]);
}
```

Do you think we need to change that? Should we create a story?

By the way, I feel this method name is not clear. It is not removing from response, it is removing FOR response in that case (doing something to help / benefit). Also, in fact it should be more generic as the documentation describes it: "In-place removal of certain keys from the input object", so it might be worth to remove the response and it helps to remove attributes for a given Object (even now we are just using to give it back to the client).

Reply:
feel free to rename the fn to make it more clear

the hierarchy response is a special case because the structure is hierarchical, where any subject can contain an array of subjects (the "children" attribute) -- *IF* we are ever removing attributes from a subject in the response from a call to `GET /v1/subjects/{key}/ hierarchy` then we need to traverse the hierarchy and remove those fields from every subject, not just the root-level subject

Reply:
the fields param only affects the top-level anyway, so we don't need to remove what wasn't added.